### DISCUSSION TITLE
Can I remove files or use .airflowignore to reduce the scheduler load

### DISCUSSION THREAD

**Discussion by CatarinaSilva** [apache/airflow#18247] [Q&A] [answered]
### Context

I have thousands of DAGs that get generated automatically and parameterize and I am managing versions by changing the filenames/dag_ids by using the pattern `test_dag_vX`.  For this purpose though, we are keeping all files available in the same folder where the active versions live.

### Question

Nothing but the latest version is meant to be triggered, so I wanted to ensure the scheduler ignores the old versions of DAGs but still have them show in the UI (for history purposes) . 

I was wondering if any of the following is a valid option:

* Can I make sure the scheduler stops looking at these files by removing them - and will the UI still work since it goes to the db? or will they be deleted from the serialized dags table and therefore fail?
* Can I use instead `.airflowignore` for this purpose?

**Comment by BasPH** [comment, marked as answer]
If you remove a DAG file, Airflow will not display it anymore in the UI, but the metadata is not removed. And if you return the DAG file, you'll see it again in the UI, with all history. `.airflowignore` essentially has the same effect.

To not run the DAG but keep it visible in the UI, your only option is to pause the DAG. You might want to look at the REST API with that number of DAGs. Paused DAGs are ignored by the scheduler, so that should reduce the stress a bit.

In your case it might help to tweak the scheduler settings a bit, have you looked into that?
